android/titanium/src/java/org/appcelerator/titanium/TiBaseActivity.java

```diff
@@ -1726,6 +1684,22 @@ protected boolean shouldFinishRootActivity()
 		return TiBaseActivity.canFinishRoot && isIntentRequestingFinishRoot;
 	}
 
+	@Override
+	public void finishAfterTransition()
```

Comment:
Is this override necessary, does't `finishAfterTransition` call `finish`?

Reply:
The override is needed to call the `removeFromActivityStack(this)` like we do in the `finish()` method... because I removed the method call from the `WindowProxy` class. Now all of the `addToActivityStack()` and `removeFromActivityStack()` calls are done in the same place, the `TiBaseActivity` class. The old code wants to remove the activity from the stack (before the `onDestroy()` happens) so that the next child `Ti.UI.Window` that the code might open won't use its context.